S3T3n1/src/main/java/org/example/model/domain/Application.java

```diff
@@ -132,41 +132,55 @@ public static int searchInvoiceLog(int id){
     }
 
     public static void removeTree() {
-        int id = removerPrompt("Tree");
-        int listIndex = searchList(id, "product");
-
-        if (flowerShop.getStockList().get(listIndex) != null) {
-            Product product = flowerShop.getStockList().get(listIndex);
-            flowerShop.removeStock(flowerShop.getStockList().get(listIndex));
-            System.out.println("The " + product.getType() + " " + product.getId() + " " + product.getName() + " has been removed from the list.\n");
-        } else {
-            System.out.println("The tree with Id# " + id + " does not exist.\n");
+        if(flowerShop.getProductMap().get("Tree") == 0){
+            System.out.println("There are no trees in stock.\n");
+        }else {
+            int id = removerPrompt("Tree");
+            int listIndex = searchList(id, "product");
+
+            if (flowerShop.getStockList().get(listIndex) != null) {
+                Product product = flowerShop.getStockList().get(listIndex);
+                flowerShop.removeStock(flowerShop.getStockList().get(listIndex));
```

Comment:
this is already product no? instead of repeating getstocklist.get(listindex)  we can save it once and use it as a value later on?